app/models/manageiq/providers/hawkular/middleware_manager/event_catcher_mixin.rb

```diff
@@ -0,0 +1,67 @@
+# We use this to provide easy access to both the runner and specs
+module ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcherMixin
+  # extend ActiveSupport::Concern
+
+  # TODO: perhaps we need to block certain events.  Another way to do this would be to add a tag on events we want to
+  # suck into MiQ, and just fetch with the tag as criteria
+  INCLUDE_EVENTS = {'Hawkular Deployment' => '.*'}.freeze # some Hash of included events
+  EXCLUDE_EVENTS = {}.freeze # TODO: some Hash of exclusions for the included events
```

Comment:
@durandom OK, I think I am understanding this, there are a couple of things in play.  A provider defines the event types that can be reported, every queued event should be of a defined type.  "Blacklisted" event types are those that are defined but should not be reported. An event type can be blacklisted by default by a provider, as shown in the above link, or set blacklisted in other ways, probably by a user or maybe also automatically to prevent flooding or something like that.  This blacklist filtering has built in support, and it is something I should add to this PR.

>  In case you want some events not to get into the system, you should do it as early as possible, i.e. right in the REST call...

The logic we're discussing falls into this category, I think.  In Hawkular the set of events is unbounded, it's not controlled as events can be injected from lots of different places. The event catcher in this provider pulls these Hawkular events and can not predict what it will get from the polling rest call.   But it is the event catcher that knows which event types are defined for the Hawkular provider.  I see that in this PR I do this filtering in process_event, that is too late, it should be done before adding the event to the queue.  I had actually changed that already in a follow-on branch, but I'll make that change here now.

So, my plan would be to add the blacklist filter support, and then also to move my provider filtering before adding to the queue, more like an extension of the rest call. Sound correct?